core/drivers/crypto/hisilicon/hisi_qm.c

```diff
@@ -143,49 +147,114 @@ static void qm_db(struct hisi_qm *qm, uint16_t qn, uint8_t cmd, uint16_t index,
 	io_write64(qm->io_base + QM_DOORBELL_SQ_CQ_BASE, doorbell);
 }
 
+static void qm_mb_write(struct hisi_qm *qm, struct qm_mailbox *mb)
+{
+	vaddr_t dst = qm->io_base + QM_MAILBOX_BASE;
+
+	write_64bit_pair(dst, mb->x[1], mb->x[0]);
+	dsb();
+}
+
+static void qm_mb_read(struct hisi_qm *qm, struct qm_mailbox *mb)
+{
+	vaddr_t mb_base = qm->io_base + QM_MAILBOX_BASE;
+	uint64_t tmp0 = 0;
+	uint64_t tmp1 = 0;
+
+	asm volatile ("ldp %0, %1, [%2]\n" : "=&r"(tmp0), "=&r"(tmp1) :
+		      "r"(mb_base) : "memory");
+	dsb();
```

Comment:
Why is this done after reading the shared mailbox? Is this `dsb()` needed at all?

Reply:
dsb() is needed to make sure that all previous data read and write operations are completed before the instruction is executed.dsb() has been changed to dsb_osh().

Reply:
Got it, thanks.